parachain/pallets/ethereum-beacon-client/src/types.rs

```diff
@@ -24,3 +24,13 @@ pub(crate) type ExecutionHeaderBuffer<T> = RingBufferMapImpl<
 	crate::ExecutionHeaders<T>,
 	OptionQuery,
 >;
+
+/// FinalizedHeader ring buffer implementation
+pub(crate) type FinalizedHeaderBuffer<T> = RingBufferMapImpl<
```

Comment:
Remember, we're storing fields from [BeaconState](https://github.com/ethereum/consensus-specs/blob/dev/specs/phase0/beacon-chain.md#beaconstate), not BeaconHeader.

```suggestion
pub(crate) type FinalizedBeaconStateBuffer<T> = RingBufferMapImpl<
```